src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -113,11 +196,38 @@
 									output += ' (' + item.englishName + ')';
 								}
 								return output;
-							}).toString() : '',
-							librarianNote: item.librarianNote
+							}).join(', ') : '',
+							librarianNote: item.librarianNote,
+							relBooks: [],
+							uploader: item.uploader.email,
+							text: item.objectId
 						};
+
+						return $scope.booksCache[item.objectId];
 					});
+
+					return bookService.getAllBookRelationships();
+				}).then(function(results) {
+					var currentBook;
```

Comment:
It might be more obvious in full context than in a diff, but I think a comment indicating what results are would help...or a more explicit name. I think each item is an array of books.
This won't scale to more than about 1000 relationships...parse.com will truncate the results...but I think we already accepted having to rework the grid view (at least) when we reach 1000 books.